cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MetricsProcessorTwillRunnable.java

```diff
@@ -94,6 +98,13 @@ protected Injector doInit(TwillContext context) {
   public void addServices(List<? super Service> services) {
     services.add(injector.getInstance(MessagingMetricsProcessorRuntimeService.class));
     services.add(injector.getInstance(MetricsProcessorStatusService.class));
+
+    // If a binding for MetricsAdminSubscriberService exists, add it as well.
```

Comment:
interesting. Is that because the instanceId is not available at this time? If it is, I would find it a little more straight-forward to always bind it, but only start it in instance 0.

Reply:
Yeah sure.